cartridge/pool.lua

```diff
@@ -193,25 +197,58 @@ local function map_call(fn_name, args, opts)
     end
 
     local retmap, errmap = {}, {}
-    local fibers = {}
-    for _, uri in pairs(opts.uri_list) do
-        local fiber = fiber.new(
-            _gather_netbox_call,
-            fiber.self().storage, retmap, errmap, uri, fn_name, args,
-            {timeout = opts.timeout}
-        )
-        fiber:name('netbox_map_call')
-        fiber:set_joinable(true)
-        fibers[uri] = fiber
+    local fibers = table.new(0, #opts.uri_list)
+    local futures = table.new(0, #opts.uri_list)
+
+    local timeout = opts.timeout or vars.options.MAP_CALL_TIMEOUT
+    local deadline = fiber.clock() + timeout
+
+    for _, uri in ipairs(opts.uri_list) do
+        local conn, err = connect(uri, {wait_connected = false})
+        if conn == nil then
+            err = err or 'Unknown error'
+            errmap[uri] = NetboxConnectError:new('%q: %s', uri, err)
+        elseif conn:is_connected() then
+            local future, err = errors.netbox_call(
+                conn, fn_name, args, {is_async = true}
+            )
+            if future == nil then
+                errmap[uri] = err
+            end
+            futures[uri] = future
```

Comment:
[Below](https://github.com/tarantool/cartridge/pull/1424/files#diff-e337ea42ddc0b628cab1c67abd93055e8ee30bc20b5eef35f4a43acd9ae6ababR247) you use just
```lua
retmap[uri] = ret
errmap[uri] = err
```

I don't see any reasons to make code less uniform.